include/elliptics/packet.h

```diff
@@ -340,30 +359,67 @@ struct dnet_backend_control
 struct dnet_id_container
 {
 	int backends_count;
-	struct dnet_backend_ids backends[0];
 } __attribute__ ((packed));
 
-static inline int dnet_validate_id_container(struct dnet_id_container *ids, size_t size, struct dnet_backend_ids **backends)
+static inline void dnet_convert_id_container(struct dnet_id_container *cnt)
```

Comment:
I reworked id container validation to fix heap overflow and le/be conversion bugs.
It is not used by clients directly, but it is part of structures which client sends to match elliptics protocol, so it lives in public headers.